index.bs

```diff
@@ -1468,19 +1512,186 @@ To <dfn>validate a suffix</dfn> |suffix|, run the following steps:
+
+A user agent holds a <dfn>recently picked directory map</dfn>, which is a
+[=map=] of [=/origins=] to [=path id maps=].
+
+A <dfn>path id map</dfn> is a [=map=] of [=valid path ids=] to paths.
+
+A <dfn>valid path id</dfn> is a [=string=] where each character is [=ASCII alphanumeric=] or "_" or "-".
+
+To prevent a [=path id map=] from growing without a bound, user agents should implement
+some mechanism to limit how many recently picked directories will be remembered. This
+can for example be done by evicting least recently used entries.
+
+The <dfn enum>WellKnownDirectory</dfn> enum lets a website pick one of several well-known
+paths. The exact paths the various values of this enum map to is left up to the user agent.
+The following list describes the meaning of each of the values, and gives possible example paths on different operating systems:
+
+<dl dfn-for=WellKnownDirectory>
+: <dfn enum-value>"desktop"</dfn>
+:: The user's Desktop directory, if such a thing exists. For example this could be
+   `C:\Documents and Settings\username\Desktop`, `/Users/username/Desktop`, or `/home/username/Desktop`.
+: <dfn enum-value>"documents"</dfn>
+:: Directory in which documents created by the user would typically be stored.
+   For example `C:\Documents and Settings\username\My Documents`, `/Users/username/Documents`, or `/home/username/Documents`.
+: <dfn enum-value>"downloads"</dfn>
+:: Directory where downloaded files would typically be stored.
+   For example `C:\Documents and Settings\username\Downloads`, `/Users/username/Downloads`, or `/home/username/Downloads`.
+: <dfn enum-value>"music"</dfn>
+:: Directory where audio files would typically be stored.
+   For example `C:\Documents and Settings\username\My Documents\My Music`, `/Users/username/Music`, or `/home/username/Music`.
+: <dfn enum-value>"pictures"</dfn>
+:: Directory where photos and other still images would typically be stored.
+   For example `C:\Documents and Settings\username\My Documents\My Pictures`, `/Users/username/Pictures`, or `/home/username/Pictures`.
+: <dfn enum-value>"videos"</dfn>
+:: Directory where videos/movies would typically be stored.
+   For example `C:\Documents and Settings\username\My Documents\My Videos`, `/Users/username/Movies`, or `/home/username/Videos`.
+
+<div algorithm>
+To <dfn>determine the directory the picker will start in</dfn>, given an optional [=string=] |id|,
+an optional {{StartInDirectory}} |startIn| and an [=environment settings object=] |environment|,
+run the following steps:
+
+1. If |id| given, and is not a [=valid path id=], throw a {{TypeError}}.
+1. If |id|'s [=string/length=] is more than 32, throw a {{TypeError}}.
+
+1. Let |origin| be |environment|'s [=environment settings object/origin=].
+
+1. If |startIn| is a {{FileSystemHandle}}:
+  1. Let |entry| be |startIn|'s [=FileSystemHandle/entry=].
+  1. If |entry| does not represent an [=/entry=] in an [=origin private file system=]:
+    1. If |entry| is a [=file entry=],
```

Comment:
Nit: Since this is an algorithm, it would be better to rephrase this from "If... return... if..." to "If... and ... then return..." if it can be done concisely. 

Here and below.

Reply:
Sorry, I'm not sure I understand what you're asking for...

Reply:
Right now the text is:

> If entry is a file entry, return the path on the local file system corresponding to the parent directory of entry, if such a path can be determined.

I parse this as: "If A, then B, if C" 

I believe the intent is "If (A and C), then B" but it could also be interpreted as "If A, then (if B then C, otherwise ???)"

I think it would be marginally clearer as something like:

> If entry is a file entry, and a path on the local file system corresponding to the parent directory of entry can be determined, then return that path.

But overall I think this is minor.

Reply:
Ah, yes, that makes a lot of sense. Done.